web/src/main/java/com/epam/rft/atsy/web/controllers/ApplicationStateController.java

```diff
@@ -1,21 +1,49 @@
 package com.epam.rft.atsy.web.controllers;
 
+import com.epam.rft.atsy.service.StatesService;
+import com.epam.rft.atsy.service.domain.states.StateDTO;
+import com.epam.rft.atsy.service.domain.states.StateViewHistoryDTO;
+import org.springframework.context.MessageSource;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.servlet.ModelAndView;
 
+import java.util.Collection;
+import java.util.Locale;
+import javax.annotation.Resource;
+
 @Controller
 @RequestMapping(value = "/secure/application_state")
 public class ApplicationStateController {
 
   private static final String VIEW_NAME = "application_state";
+  private static final String APPLICATION_STATE = "candidate.table.state.";
+  private static final String STATES_OBJECT_KEY = "states";
+  @Resource
+  private StatesService statesService;
+
+  @Resource
+  private MessageSource messageSource;
 
   @RequestMapping(method = RequestMethod.GET)
-  public ModelAndView loadPage(@RequestParam Long applicationId) {
+  public ModelAndView loadPage(@RequestParam Long applicationId, Locale locale) {
     ModelAndView modelAndView = new ModelAndView(VIEW_NAME);
     modelAndView.addObject("applicationId", applicationId);
+
+    Collection<StateViewHistoryDTO>
+        applicationStates =
+        statesService.getStatesByApplicationId(applicationId);
+
+    for (StateViewHistoryDTO stateViewHistoryDTO : applicationStates) {
```

Comment:
This is the point where it is better to have a new class in the web layer which contains the state history view from the ui perspective.

You can convert the StateHistoryViewDTO instances into the StateHistoryViewXY object and put in the necessary messages.